### ISSUE TITLE
Handle Service Unavailability by Throwing an Error

### ISSUE THREAD

**Issue by keviny2** [ivanp1994/PyMart#7]
## Description

Currently, the package does not handle the case where the target service returns an HTML error page (e.g., "Service Unavailable") instead of the expected XML content. This results in a ParseError when attempting to parse the HTML content as XML using ElementTree.fromstring(). The error message typically looks like this:

`ParseError: mismatched tag: line 62, column 2`

This can be confusing for users, as the underlying issue is that the service is unavailable, not a problem with the XML structure itself.

## Desired Behavior
When the service returns an HTML error page, the package should:

1. Detect that the response is HTML and not XML.
2. Throw a clear and descriptive error indicating that the service is temporarily unavailable.

**Comment by ivanp1994** [comment]
Yeah, this works, I've integrated it in the base.py (I dont think a separate file is needed for errors, esp when errors are just one class) and pushed. Thanks a lot!
